crates/primitives/src/chain_spec.rs

```diff
@@ -154,6 +173,68 @@ impl ChainSpec {
     }
 }
 
+impl From<EthersGenesis> for ChainSpec {
+    fn from(genesis: EthersGenesis) -> Self {
+        let alloc = genesis
+            .alloc
+            .iter()
+            .map(|(addr, account)| (addr.0.into(), account.clone().into()))
+            .collect::<HashMap<H160, GenesisAccount>>();
+
+        let state_root = genesis_state_root(alloc.clone());
+
+        let genesis_block = Genesis {
+            nonce: genesis.nonce.as_u64(),
+            timestamp: genesis.timestamp.as_u64(),
+            gas_limit: genesis.gas_limit.as_u64(),
+            difficulty: genesis.difficulty.into(),
+            mix_hash: genesis.mix_hash.0.into(),
+            coinbase: genesis.coinbase.0.into(),
+            extra_data: genesis.extra_data.0.into(),
+            alloc,
+            state_root,
+        };
+
+        let genesis_hash = Header::from(genesis_block.clone()).seal().hash();
+        let paris_ttd = genesis.config.terminal_total_difficulty.map(|ttd| ttd.into());
+        let hardfork_opts = vec![
+            (Hardfork::Homestead, genesis.config.homestead_block),
+            (Hardfork::Dao, genesis.config.dao_fork_block),
+            (Hardfork::Tangerine, genesis.config.eip150_block),
+            // TODO: eip-158 was also activated when eip-155 was activated, but we don't have the
+            // proper hardfork identifier for it. this breaks the hardfork abstraction slightly
+            (Hardfork::SpuriousDragon, genesis.config.eip155_block),
+            (Hardfork::Byzantium, genesis.config.byzantium_block),
+            (Hardfork::Constantinople, genesis.config.constantinople_block),
+            (Hardfork::Petersburg, genesis.config.petersburg_block),
+            (Hardfork::Istanbul, genesis.config.istanbul_block),
+            (Hardfork::Muirglacier, genesis.config.muir_glacier_block),
+            (Hardfork::Berlin, genesis.config.berlin_block),
+            (Hardfork::London, genesis.config.london_block),
+            (Hardfork::ArrowGlacier, genesis.config.arrow_glacier_block),
+            (Hardfork::GrayGlacier, genesis.config.gray_glacier_block),
+            // TODO: similar problem as eip-158, but with the merge netsplit block. only used in
+            // sepolia, but required for proper forkid generation
```

Comment:
What is the fix here?

Reply:
The first TODO:
Using `Hardfork::Tangerine` works for mainnet because 155/158 were both activated in tangerine, to fix this for an arbitrary genesis we might want to split `Hardfork::Tangerine` into:
```rust
Hardfork::Eip150,
Hardfork::Eip158,
```

The second TODO:
I actually added `Hardfork::MergeNetsplit`, so we just need to add
```rust
            (Hardfork::MergeNetsplit, genesis.config.merge_netsplit_block),
```
and delete the TODO

Reply:
SGTM  - easy followup PR